### PR TITLE
Add exception message when logging resolved exceptions

### PR DESCRIPTION
**Pull request by LucaBarden**
For more insight into which exception has been resolved, the thrown exception will be included in the log message accordingly

### CODE REVIEW

**Comment by Duckulus** [comment]
Huge W

**Review by rstoyanchev — COMMENTED** [review]
The warnLogger is intentionally minimal, and may be used in production for example. At most we could add the Exception message in addition to the Exception name. Logging the full exception is not in scope I'm afraid. You can, however, override the `buildLogMessage` method.

**Comment by LucaBarden** [comment]
> The warnLogger is intentionally minimal, and may be used in production for example. At most we could add the Exception message in addition to the Exception name. Logging the full exception is not in scope I'm afraid. You can, however, override the `buildLogMessage` method.

We are using it in production and have searched for issues too long most of the time and would have appreciated the Stacktrace included.

**Comment by bclozel** [comment]
I agree with Rossen's assessment. A large part of the Spring community will be concerned by logs becoming too noisy. I'm sure lots of teams do not use Kibana or similar and will be negatively impacted by this change.

Please update your PR to log the exception message in the default implementation of `buildLogMessage`.

**Comment by LucaBarden** [comment]
Let me know if you had a different style in mind

File: spring-webmvc/src/main/java/org/springframework/web/servlet/handler/AbstractHandlerExceptionResolver.java [hunk 1]
```diff
@@ -255,7 +255,7 @@ protected void logException(Exception ex, HttpServletRequest request) {
 	 * @return the log message to use
 	 */
 	protected String buildLogMessage(Exception ex, HttpServletRequest request) {
-		return "Resolved [" + LogFormatUtils.formatValue(ex, -1, true) + "]";
+		return "Resolved [" + LogFormatUtils.formatValue(ex, -1, true) + "] :" + ex.getMessage();
```
Comment (rstoyanchev) [inline spring-webmvc/src/main/java/org/springframework/web/servlet/handler/AbstractHandlerExceptionResolver.java:258]: This should be in the formatValue call:
```suggestion
		return "Resolved [" + LogFormatUtils.formatValue(ex + "] :" + ex.getMessage(), -1, true);
```

**Comment by rstoyanchev** [comment]
Thanks for the updates. Apologies for any confusion, but on closer look, `Throwable#toString` [already does this](https://github.com/openjdk/jdk/blob/9ebc2ecbf613da3bcee1dd5e8920a26d5f6d6df7/src/java.base/share/classes/java/lang/Throwable.java#L523-L527), so that's about as far as we can take it. If you want full exception logging, you can override `buildLogMessage` . 

Note that the Spring MVC Auto-Configuration in Spring Boot makes it easy to replace the `ExceptionHandlerExceptionResolver` instance through a `WebMvcRegistrations` bean. See [the docs on that](https://docs.spring.io/spring-boot/reference/web/servlet.html#web.servlet.spring-mvc.auto-configuration).

**Pull request closed without merging**